cstrml/market/src/lib.rs

```diff
@@ -336,7 +336,13 @@ decl_module! {
             pledge_ledger.total -= value;
 
             // 5 Upsert pledge ledger
-            Self::upsert_pledge_ledger(&who, &pledge_ledger);
+            if pledge_ledger.total.is_zero() && pledge_ledger.used.is_zero() {
```

Comment:
To my understanding, if total = 0, used should be 0, right? So the 2nd condition is useless?

Reply:
fixed.

Reply:
To make 5 step robust, two checks is ok. Relying on the previous check, the 2nd condition is useless. Remove it in the consideration of efficiency